src/prefect/orion/api/flow_runs.py

```diff
@@ -180,6 +181,16 @@ async def read_flow_runs(
         sort=sort,
     )
 
+    # Instead of relying on fastapi.encoders.jsonable_encoder to convert the
+    # response to JSON, we do so more efficiently ourselves.
+    # In particular, the FastAPI encoder is very slow for large, nested objects.
+    # See: https://github.com/tiangolo/fastapi/issues/1224
+    encoded = [
+        schemas.core.FlowRun.from_orm(fr).dict(json_compatible=True)
+        for fr in db_flow_runs
+    ]
+    return ORJSONResponse(content=encoded)
```

Comment:
To @madkinsz point, how about this as a compromise:

```suggestion
    return cast(List[schemas.core.FlowRun], ORJSONResponse(content=encoded))
```

Reply:
Zach restored the original type annotation on the function itself. We'll probably get a `mypy` error now but we can tackle that and casting separately when actually start running type checks here?